airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -70,75 +71,83 @@ class SnowflakeDestinationHandler(
     private fun getFinalTableRowCount(
         streamIds: List<StreamId>
     ): LinkedHashMap<String, LinkedHashMap<String, Int>> {
-        val tableRowCounts = LinkedHashMap<String, LinkedHashMap<String, Int>>()
-        // convert list stream to array
-        val namespaces = streamIds.map { it.finalNamespace }.toTypedArray()
-        val names = streamIds.map { it.finalName }.toTypedArray()
-        val query =
-            """
-            |SELECT table_schema, table_name, row_count
-            |FROM information_schema.tables
-            |WHERE table_catalog = ? 
-            |AND table_schema IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-            |AND table_name IN (${IntRange(1, streamIds.size).joinToString { "?" }})
-            |""".trimMargin()
-        val bindValues = arrayOf(databaseName) + namespaces + names
-        val results: List<JsonNode> = database.queryJsons(query, *bindValues)
-        for (result in results) {
-            val tableSchema = result["TABLE_SCHEMA"].asText()
-            val tableName = result["TABLE_NAME"].asText()
-            val rowCount = result["ROW_COUNT"].asInt()
-            tableRowCounts
-                .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }[tableName] =
-                rowCount
+
+        val tableRowCountsFromShowQuery = LinkedHashMap<String, LinkedHashMap<String, Int>>()
+        var showColumnsResult: List<JsonNode> = listOf()
+
+        try {
+            for (stream in streamIds) {
+                val showColumnsQuery =
+                    String.format(
+                            """
+                                SHOW TABLES LIKE '%s' IN "%s"."%s";
+                            """.trimIndent(),
+                            stream.finalName,
+                            databaseName,
+                            stream.finalNamespace,
+                            )
+                showColumnsResult = database.queryJsons(
+                    showColumnsQuery,
+                )
+                for (result in showColumnsResult) {
+                    val tableSchema = result["schema_name"].asText()
+                    val tableName = result["name"].asText()
+                    val rowCount = result["rows"].asText()
+
+                    tableRowCountsFromShowQuery
```

Comment:
The indentation is super confusing here (probably enforced by our format command). Any way to change that, or is our formatter going to bark at you?
Also, you can simplify with map`.computeIfAbsent(tableSchema) { LinkedHashMap() }`

Reply:
I think we can also use a linkedlist.withDefault, which would simplify this further